src/dmd/compiler.d

```diff
@@ -12,7 +12,143 @@
 
 module dmd.compiler;
 
+import core.stdc.stdio;
+import dmd.cond;
+import dmd.globals;
+
 struct Compiler
 {
     const(char)* vendor; // Compiler backend name
 }
+
+/**
+ * Add default `version` identifier for dmd, and set the
+ * target platform in `global`.
+ * https://dlang.org/spec/version.html#predefined-versions
+ *
+ * Needs to be run after all arguments parsing (command line, DFLAGS environment
+ * variable and config file) in order to add final flags (such as `X86_64` or
+ * the `CRuntime` used).
+ */
+void addDefaultVersionIdentifiers()
+{
+    VersionCondition.addPredefinedGlobalIdent("DigitalMars");
+    static if (TARGET_WINDOS)
+    {
+        VersionCondition.addPredefinedGlobalIdent("Windows");
+        global.params.isWindows = true;
```

Comment:
This line should really not be here. "Setting predefined identifiers" and "setting the target platform in global" should be in different functions.